### ISSUE TITLE
ZManaged integration improvements

### ISSUE THREAD

**Issue by jdegoes** [zio/zio#3880]
These could be added in the JVM (platform-specific) trait that is mixed into the `ZManaged` companion object:

- [x] `ZManaged.readFile(path: Path): ZManaged[Blocking, IOException, FileInputStream]`
- [x] `ZManaged.readFile(path: String): ZManaged[Blocking, IOException, FileInputStream]`
- [x] `ZManaged.writeFile(path: Path): ZManaged[Blocking, IOException, FileOutputStream]`
- [x] `ZManaged.writeFile(path: String): ZManaged[Blocking, IOException, FileOutputStream]`
- [x] `ZManaged.readURL(url: URL): ZManaged[Blocking, IOException, InputStream]`
- [x] `ZManaged.readURL(url: String): ZManaged[Blocking, IOException, InputStream]`
- [x] `ZManaged.readURI(uri: URI): ZManaged[Blocking, IOException, InputStream]`

```scala
final case class FileInputStream private ( private ... ) {
  def readN(n: Int): ZIO[Blocking, IOException, Option[Chunk[Byte]]]
  def readAll: ZIO[Blocking, IOException, Option[Chunk[Byte]]]
  def seekTo(n: Long): ZIO[Blocking, IOException, Unit] // look / copy FIS
}
final case class InputStream private ( private ... ) {
  def readN(n: Int): ZIO[Blocking, IOException, Option[Chunk[Byte]]]
  def readAll: ZIO[Blocking, IOException, Option[Chunk[Byte]]]
}
final case class OutputStream private ( private ... ) {
  def write(chunk: Chunk[Byte]): ZIO[Blocking, IOException, Unit]
}
```

/cc @adrianfilip

**Comment by adrianfilip** [comment]
Thanks! I'll implement this!

**Comment by iravid** [comment]
Resolved in #3901.
